meson.build

```diff
@@ -2225,6 +2225,23 @@ if conf.get('HAVE_SYSV_COMPAT') == 1
                    install_dir : systemgeneratordir)
 endif
 
+executable('systemd-xdg-autostart-generator',
+           'src/xdg-autostart-generator/xdg-autostart-generator.c',
+           include_directories : includes,
+           link_with : [libshared],
+           install_rpath : rootlibexecdir,
+           install : true,
```

Comment:
We probably want an option in `meson_options.txt` for this.

Reply:
Yup.

btw. if there is an easy way to runtime detect whether the generator is available, then that could be useful. If not, we'll just tell distributions to only enable the feature in the DEs when systemd is known to provide the generator.

Reply:
My first thought was to say: just do the equivalent of `[ test -x /usr/lib/systemd/user-generator/foobar ]`, possibly using systemd-path or pkgconfig systemd to query the directory. But it's not so simple, because we have multiple paths, and generators may be masked. So the answer to the question "is the generator installed" is easy, but the answer to "is the generator active" would be more complicated, because it'd require looping over some directories. I guess we could add an api to get the "active generator list".

Reply:
To be honest, then I'll just add a variable to the pkgconfig file and we can pick that up to enable/disable the use of the generator at compile time.

And yeah, I really don't think I care much about someone manually masking it. They should be aware of the possibility of shooting themselves in the foot. It is not like we advertise the possibility of doing this.

Thanks for the answer!